src/sdam/description/server.rs

```diff
@@ -122,6 +122,7 @@ impl PartialEq for ServerDescription {
 
                 self_response == other_response
             }
+            (Err(self_err), Err(other_err)) => self_err == other_err,
```

Comment:
It looks like changes from some recent commits got included in the diff for this PR. Can you rebase your branch on main to fix? (let me know if you need any help with this)